src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
If one has `NUMBER(3,0)` which is our of the `int8` range - one will have to use a bigger precision.
It will be done explicitly most of the time.

Also in a situation when there's no metadata for aggregation columns, plus there are cases when one is summing up `NUMBER(3,0)`s it could theoretically result in a number outside of the precision of the components.

Thus in any case there's a work-around - manual cast with elected precision.
Seems like a better alternative than to loose half of the range available of values.